ejb/WFLY-12190-http-server-to-server-config.adoc

```diff
@@ -0,0 +1,153 @@
+= EJB over HTTP server-to-server invocation
+:email:             tomasz.adamski@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+=== Issue
+* https://issues.jboss.org/browse/WFLY-12190
+
+=== Dev Contacts
+
+* mailto:tadamski@redhat.com
+
+=== QE Contacts
+
+=== Testing By
+[ ] Engineering
+
+[x] QE
+
+=== Affected Projects or Components
+EJBs
+
+== Introduction
+
+PRODMGT-268 (https://issues.redhat.com/browse/PRODMGT-268) added the ability to configure EJB invocations over HTTP protocol. Customers would like to use this feature in server-to-server configurations.
+
+== Proposed configuration
+
+ It was requested to modify server configuration so that http protocol can be used from within remote outbound connections. Sample configuration would be as follows:
+[source]
+----
+<remote-outbound-connection name="http-connection" outbound-socket-binding-ref="binding-remote-ejb-connection" protocol="http"/>
+----
+
+Such connection can be then used from within jboss-ejb-client.xml configuration file. The following issues have to be taken into consideration regarding such configuration:
+
+=== Http config in remoting subystem
+
+Remoting subsystem is responsible for the configuration of connection based on jboss-remoting protocol. The attributes and properties of remote-outbound-connections are used to configure remoting connections and make no sense in for http protocol. Simply adding another protocol to remote-outbound-connection may seem like a simplest solution from the customer point of view, but it introduces incoherent subsystem configuration and in my opinion it is unnaceptable.
```

Comment:
I see the incoherency in subsystem configuration just I can imagine the headache for users configuring the different outbound connection on different places.
Should not be then deprecated the usage of the `remote-outbound-connections` in `remoting` subsystem and document that the correct place for configuration of any protocol is the `ejb3` subsystem from now?

Reply:
@ochaloup I think that such configuration is correct. Remoting connections have their own configuration and IMO it's place is in remoting subsystem. This is a bit elaborate config but it was always the case.

Reply:
ok, I see